route/kafkamdm.go

```diff
@@ -0,0 +1,240 @@
+	// Because we don't change the flush settings, sarama will try to produce messages
+	// as fast as possible to keep latency low.
+	config := sarama.NewConfig()
+	config.Producer.RequiredAcks = sarama.WaitForAll // Wait for all in-sync replicas to ack the message
+	config.Producer.Retry.Max = 10                   // Retry up to 10 times to produce the message
+	config.Producer.Compression, err = getCompression(codec)
+	if err != nil {
+		log.Fatal(5, "kafkaMdm %q: %s", r.key, err)
+	}
+	config.Producer.Return.Successes = true
+	config.Producer.Timeout = time.Duration(timeout) * time.Millisecond
+	err = config.Validate()
+	if err != nil {
+		log.Fatal(4, "kafkaMdm %q: failed to validate kafka config. %s", r.Key, err)
+	}
+	r.saramaCfg = config
+
+	r.config.Store(baseConfig{*m, make([]*dest.Destination, 0)})
+	go r.run()
+	return r, nil
+}
+
+func (r *KafkaMdm) run() {
+	metrics := make([]*schema.MetricData, 0, r.flushMaxNum)
+	ticker := time.NewTicker(r.flushMaxWait)
+	brokers := []string{r.broker}
+
+	connect := func() {
+		var err error
+		r.producer, err = sarama.NewSyncProducer(brokers, r.saramaCfg)
+		if err == sarama.ErrOutOfBrokers {
+			log.Warning("kafkaMdm %q: %s", r.key, err)
+		} else if err != nil {
+			log.Fatal(4, "kafkaMdm %q: failed to initialize kafka producer. %s", r.key, err)
+		} else {
+			log.Notice("kafkaMdm %q: now connected to kafka", r.key)
+		}
+	}
+
+	// flushes the data to kafka and resets buffer.  blocks until it succeeds
+	flush := func() {
+		for r.producer == nil {
+			connect()
+			time.Sleep(time.Second)
+		}
+		for {
+			pre := time.Now()
+			var err error
+			size := 0
+
+			payload := make([]*sarama.ProducerMessage, len(metrics))
+
+			for i, metric := range metrics {
+				var data []byte
+				data, err = metric.MarshalMsg(data[:])
+				if err != nil {
+					panic(err)
+				}
+				size += len(data)
+
+				key, err := r.partitioner.GetPartitionKey(metric, nil)
+				if err != nil {
+					panic(err)
+				}
+				payload[i] = &sarama.ProducerMessage{
+					Key:   sarama.ByteEncoder(key),
+					Topic: r.topic,
+					Value: sarama.ByteEncoder(data),
+				}
+			}
+			err = r.producer.SendMessages(payload)
+
+			diff := time.Since(pre)
+			if err == nil {
+				log.Info("KafkaMdm %q: sent %d metrics in %s - msg size %d", r.key, len(metrics), diff, size)
+				r.numOut.Inc(int64(len(metrics)))
+				r.tickFlushSize.Update(int64(size))
+				r.durationTickFlush.Update(diff)
+				metrics = metrics[:0]
+				break
+			}
+			r.numErrFlush.Inc(1)
+			log.Warning("KafkaMdm %q: failed to submit data: %s will try again in %s (this attempt took %s)", r.key, err, diff)
+
+			time.Sleep(100 * time.Millisecond)
+		}
+	}
+	for {
+		select {
+		case buf, ok := <-r.buf:
+			if !ok {
+				if len(metrics) != 0 {
+					flush()
+				}
+				return
+			}
+			r.numBuffered.Dec(1)
+			md, err := parseMetric(buf, r.schemas)
+			if err != nil {
+				log.Error("KafkaMdm %q: %s", r.key, err)
+				continue
+			}
+			md.SetId()
```

Comment:
md has the default OrgId of 1,  you need to set it to r.orgId before calling SetId